src/coreclr/scripts/superpmi.py

```diff
@@ -1285,168 +1343,53 @@ def replay_with_asm_diffs(self):
-                        # as the loadlibrary path will be relative to the current directory.
-                        with ChangeDir(self.coreclr_args.core_root) as dir:
-                            # Generate diff and base asm
-                            base_txt = None
-                            diff_txt = None
-
-                            command = [self.superpmi_path] + flags + [self.base_jit_path, mch_file]
-
-                            base_asm_path = os.path.join(base_asm_location, "{}.dasm".format(item))
-                            with open(base_asm_path, 'w') as file_handle:
-                                logging.debug("{}Generating {}".format(print_prefix, base_asm_path))
-                                logging.debug("{}Invoking: {}".format(print_prefix, " ".join(command)))
-                                proc = await asyncio.create_subprocess_shell(" ".join(command), stdout=file_handle, stderr=asyncio.subprocess.PIPE)
-                                await proc.communicate()
-
-                            with open(base_asm_path, 'r') as file_handle:
-                                base_txt = file_handle.read()
-
-                            command = [self.superpmi_path] + flags + [self.diff_jit_path, mch_file]
-
-                            diff_asm_path = os.path.join(diff_asm_location, "{}.dasm".format(item))
-                            with open(diff_asm_path, 'w') as file_handle:
-                                logging.debug("{}Generating {}".format(print_prefix, diff_asm_path))
-                                logging.debug("{}Invoking: {}".format(print_prefix, " ".join(command)))
-                                proc = await asyncio.create_subprocess_shell(" ".join(command), stdout=file_handle, stderr=asyncio.subprocess.PIPE)
-                                await proc.communicate()
-
-                            with open(diff_asm_path, 'r') as file_handle:
-                                diff_txt = file_handle.read()
-
-                            # Sanity checks
-                            assert base_txt is not None
-                            assert base_txt != ""
-
-                            assert diff_txt is not None
-                            assert diff_txt != ""
-
-                            if base_txt != diff_txt:
-                                text_differences.put_nowait(item)
-                    ################################################################################################ end of create_asm()
-
-                    async def create_jit_dump(print_prefix, item, self, mch_file, jit_dump_differences, base_dump_location, diff_dump_location):
-                        """ Run superpmi over an mc to create JIT dumps for the method.
+                    async def create_replay_artifacts(print_prefix, item, self, mch_file, env_vars, jit_differences_queue, base_location, diff_location, extension):
```

Comment:
there is usually a rule for long lines.

Reply:
Yes, I disabled the long lines rule... too many cases :-)